src/Browser.php

```diff
@@ -725,6 +726,49 @@ public function withResponseBuffer($maximumSize)
         ));
     }
 
+    /**
+     * The `withHeader(string $header, string $value): Browser` method can be used to
+     * add a request header for all following requests.
+     *
+     * Note that the new header will overwrite any headers previously set with the same name (case-insensitive). Following requests will use these headers by default unless they are explicitly set for any requests.
+     *
+     * @param string $header
+     * @param string $value
+     * @return Browser
+     */
+    public function withHeader($header, $value)
+    {
+        $browser = $this->withoutHeader($header);
+        $browser->defaultHeaders[$header] = $value;
+
+        return $browser;
+    }
+
+    /**
+     *
+     * The `withoutHeader(string $header): Browser` method can be used to
+     * remove any default request headers previously set via the [`withHeader()` method](#withheader).
+     *
+     * Note that this method only affects the headers which were set with the method `withHeader(string $header, string $value): Browser`
+     *
+     * @param string $header
+     * @return Browser
+     */
+    public function withoutHeader($header)
+    {
+        $browser = clone $this;
+
+        /** @var string|int $key */
+        foreach (\array_keys($browser->defaultHeaders) as $key) {
+            if (\strcasecmp($key, $header) === 0) {
+                unset($browser->defaultHeaders[$key]);
+                break;
+            }
+        }
```

Comment:
Would using `array_key_exists` here be more performant?

Reply:
Yes it should be but I'm not sure what should be achieved here. If `withoutHeader('user-agent')` removes a previously added header with `withHeader('User-Agent')` this would be correct because `array_key_exists` is not case-insensitive.

Reply:
It would be more performant, but the header could be case-insensitive.